adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -335,21 +344,60 @@ public AuthenticationResult getAuthTokenInBackground(final AuthenticationRequest
             }
 
             Logger.v(TAG, "Returning result from broker");
-            return authResult;
+            return bundleResult;
         } else {
             Logger.v(TAG, "Target account is not found");
         }
 
         return null;
     }
 
+    private boolean isBrokerAccountServiceSupported() {
+        final Intent brokerAccountServiceIntent = BrokerAccountServiceHandler.getIntentForBrokerAccountService(mContext);
+        return isServiceSupported(mContext, brokerAccountServiceIntent);
+    }
+
+    private boolean isServiceSupported(final Context context, final Intent intent) {
+        if (intent == null) {
+            return false;
+        }
+
+        final PackageManager packageManager = context.getPackageManager();
+        final List<ResolveInfo> infos = packageManager.queryIntentServices(intent, 0);
+        return infos != null && infos.size() > 0;
```

Comment:
why? we're passing the intent, if the service is not supported, queryIntentServices won't return you back anything. 

---
In reply to: [95735693](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/734#discussion_r95735693) [](ancestors = 95735693)